### PR TITLE
Remove the CDS_COMPILE_JUST_THIS define from the ConcurrentDictionary.cs file

### PR DESCRIPTION
**Pull request by AlexGhiondea**
We are not going to build that file stand-alone anymore.

### CODE REVIEW

**Comment by AlexGhiondea** [comment]
@stephentoub -- I think you already reviewed this on the TFS side.

File: src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs [hunk 1]
```diff
@@ -1987,14 +1973,7 @@ private ReadOnlyCollection<TValue> GetValues()
         [Conditional("DEBUG")]
         private void Assert(bool condition)
```
Comment (stephentoub) [inline src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs]: Should we change the call sites to just use Contract.Assert now that this Assert method is just a wrapper for Contract.Assert?

File: src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs [hunk 2]
```diff
@@ -11,29 +11,20 @@
...
 using System.Threading;
 using System.Runtime.InteropServices;
 using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Diagnostics.CodeAnalysis;
 using System.Collections;
 using System.Runtime.Serialization;
 using System.Security;
 using System.Security.Permissions;
 using System.Collections.ObjectModel;
```
Comment (stephentoub) [inline src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs]: Consider sorting the list of using, as long as you're modifying it.

**Comment by stephentoub** [comment]
LGTM

File: src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs (hunk 1 above)
Comment (AlexGhiondea) [inline src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs]: We can. I was trying to keep the code changes to a minimum though.

File: src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs (hunk 2 above)
Comment (AlexGhiondea) [inline src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs]: Done.

File: src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs (hunk 1 above)
Comment (AlexGhiondea) [inline src/mscorlib/src/System/Collections/Concurrent/ConcurrentDictionary.cs]: I have decided against doing that -- I don't think that the code churn to desktop is worth it.

**Comment by jkotas** [comment]
LGTM

**Pull request merged**